.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+    branches: [staging]
+  pull_request:
+    branches: [staging]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CHROME_BIN: /usr/bin/google-chrome
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  test:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
+      - name: Install Dependenices
+        run: npm install
+      - name: Run Test with CodeCov
+        env:
+          CHROME_BIN: /usr/bin/google-chrome
+        run: npm run test:codecov
+      - name: Run Lint & Prettier Tests
+        run: |
+          npm run lint:type
+          npm run prettier:check
+      - name: Build staging
+        run: npm run build:stage
+      - name: build production
+        run: npm run build:prod
+      - name: Code coverage test
+        if: success()
+        uses: codecov/codecov-action@v1.1.1
+
+  deployStaging:
+    name: Deploy Staging
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/staging'
```

Comment:
this should also make sure it's a push event and not a pull_request event.

This also applies to the `deployProduction` job

Reply:
Updated the IF statement to ensure it's a push event on deploying to Staging or Production.